histoqc/wsihandles/__init__.py

```diff
@@ -0,0 +1,2 @@
+from .OpenSlideHandle import *
+from .DicomHandle import *
```

Comment:
need to be careful with this approach as it creates a tight coupling to a new dependency, wsidicom

if wsidicom is not installed, even if the user doesn't try to use this on a DICOM image, it will still crash i think?

there are a few things I'm thinking:

1) what is the dependency size of WSIdicom? for example, if it is 10GB of installation that is one thing, if it is 100k that is another, the latter of which justifies adding in wsidicom into the requirements.txt

2) if it is a larger install, we may not want to assume the user has it, and thus only users which enable dicom support should have access to it. this can be done via e.g., the config file receiving a list of the all the handlers to use when reading WSI, this would thus extend to other reading libraries, like bioformats

3) i think one way of doing this is that there is a list created of each of the WSI handlers, in the order specified by the user, for example, in the config file

-config 
wsi_handlers= "openslidehandler","dicom","type3","type4..." 

-code
for wsi_handler in wsi_handlers:
          osh = wsi_handler.open(fname)
          if osh
                     return osh
if osh == None:
          error, no handler could read this image



notice that this logic is a bit more clean, because each handler gets to see the image (in the order specified by the user) and gets to choose if it wants to load the image or not, so it is the handler programmer that writes the code. in the version here, you see that the same concept code ("is directory" function) is one level higher, making it the main branch developers responsibility to integrate as opposed to the handler developers  job.

the interesting point is that in this manner, one need only import the libraries that the user specifies, breaking the tight coupling of having to install everything. 

for our purposes this is likely fine its only 2 handlers, and the dependencies are minimal, but for a larger scale project with more developers, or more handlers, the way above is definitely more correct. 

do you see the difference?  perhaps you want to take a stab at refactoring that way for your own understanding, i think it shouldb't be more than a couple minutes of effort

Reply:
did you see this comment as well?